lib/driver-cli.js

```diff
@@ -0,0 +1,265 @@
+import _ from 'lodash';
+import { KNOWN_DRIVERS, validateDriversJson, writeInstalledDriver,
+         INSTALL_TYPE_LOCAL, INSTALL_TYPE_NPM, INSTALL_TYPE_GIT,
+         INSTALL_TYPE_GITHUB } from './drivers';
+import { getLatestVersion, installPackage, linkPackage } from './npm';
+import semver from 'semver';
+
+let INSTALLED_DRIVERS;
+
+function errAndQuit (json, msg) {
+  if (json) {
+    console.log(JSON.stringify({error: msg}));
+  } else {
+    console.error(msg.red);
+  }
+  process.exit(1);
+}
+
+function log (json, msg) {
+  !json && console.log(msg);
+}
+
+async function runDriverCommand (args) {
+  let jsonResult = null;
+  try {
+    INSTALLED_DRIVERS = validateDriversJson();
```

Comment:
we probably would need some command, like reset, which is going to recreate/resync drivers json if it was damaged/edited manually. With the current approach there is no way to launch Appium if this config is damaged

Reply:
i've added a todo for this, my thought was as follows:

```
# verify driver.json and that every driver referred to there can be required OK
# just print out the status
appium driver verify

# perform verification but then reinstall anything which couldn't be required
appium driver verify --reinstall
```

Reply:
would it fix a problem with damaged config?

Reply:
what kind of config problems could be fixable automatically? here are the kinds of problems i can think of:

1. driver is mentioned in config but is not npm installed. this case would be fixed with `--reinstall`
2. driver is npm installed but not mentioned in config. the only way to fix this automatically would be to require the package.json of every module in the appium home dir, and check for appium driver metadata. this seems ... slow. also, what if it was npm linked in instead of installed? or simply copied in? there's no way to guarantee that a package in node_modules can be re-installed via NPM.
3. config for driver has a missing key. how could we automatically generate the value for a missing key?
4. config for driver has an invalid value for a key. how could we decide how to fix it automatically?
5. config for driver has an additional key beyond the valid ones. this could be removed automatically, but what's the point?

Can you think of any other "damaged" cases and how we might fix them? I can't, which is why it seemed a simple reinstall of missing modules is the best we can do.

Reply:
I just wanted to make it simple at the beginning. I mean it probably would be enough to go without config for the MVP and simply use npm for packages control.